Shelldriver: make destination authorized keys file an arguement

pmelange opened the pull request:
**Description**

Change the ShellDriver so that the authorized_keys file can be set to something other than ```~/.ssh```.  This is necessary if the DUT is running dropbear, where the authorized_keys file is located (on OpenWrt) in ```/etc/dropbear/authorized_keys```.

The default value for the new arguement is ```~/.ssh``` and therefore any old code should not need to be updated.  In addition, the call to put_ssh_key() has a new optional arg where the user can set the destination authorized_keys file.  This could also be useful for setting up a multi-user system and to put the authorized_keys file for each user in their own home directory.

<!---
This checklist roughly outlines the steps for new features, remove and add tasks as needed:
--->
**Checklist**
- [x] Documentation for the feature
- [x] The arguments and description in doc/configuration.rst have been updated
- [x] PR has been tested

Fixes: #1626

pmelange committed:
ShellDriver: add optional arg dest_authorized_keys

Add an addition optional arg "dest_authorized_keys" with the default
value of ```~/.ssh/authorized_keys```,

An example where this might be used is when an embedded system uses
dropbear as it's ssh server daemon.  In such a case, the authorized_keys
file is ```/etc/dropbear/authorized_keys```

Additionally, the ```put_key_file``` takes an optional
```dest_authorized_keys``` function parameter which defaults to the above arg.

Fixes: ##1626

Signed-off-by: Perry Melange <isprotejesvalkata@gmail.com>

Emantor commented on labgrid/driver/shelldriver.py:
```diff
@@ -258,35 +262,37 @@ def _put_ssh_key(self, keyfile_path):
     @Driver.check_active
-    def put_ssh_key(self, keyfile_path):
-        self._put_ssh_key(keyfile_path)
+    def put_ssh_key(self, keyfile_path, dest_authorized_keys = None):
```
While moving the documentation comment, please extend it for all arguments of the function.

pmelange replied:
updated

pmelange committed:
ShellDriver: Fix commit 2c62fd6 documentation comment

Move the documentation comment from _put_ssh_key to put_ssh_key along
with adding documentation for the aguements.

Documention needful of reworking introduced in 2c62fd6

Signed-off-by: Perry Melange <isprotejesvalkata@gmail.com>

pmelange commented on labgrid/driver/shelldriver.py:
```diff
@@ -217,8 +220,8 @@ def _write_key(self, keyline, dest):
-    def _put_ssh_key(self, keyfile_path):
+    @step(args=['keyfile_path', 'dest_authorized_keys'])
+    def _put_ssh_key(self, keyfile_path, dest_authorized_keys):
         """Upload an SSH Key to a target"""
```
updated

Emantor commented on labgrid/driver/shelldriver.py:
```diff
@@ -258,35 +261,44 @@ def _put_ssh_key(self, keyfile_path):
         self.logger.debug("Key not on target and not writeable, using bind mount...")
         self._run_check('mkdir -p -m 700 /tmp/labgrid-ssh/')
-        self._run("cp -aL ~/.ssh/* /tmp/labgrid-ssh/")
+        self._run(f"""cp -aL {dest_authorized_keys_dir}/* /tmp/labgrid-ssh/""")
```
Why do we need to use the triple `"` syntax here?
